### PR TITLE
HWKINVENT-160 | collectionInterval to Metric and MetricTypes

### PR DESCRIPTION
**Pull request by pavolloffay**
- Added interval field to MetricType and Metric.  
- Interval represents period in seconds at which metrics are collected. 
- In MetricType this attribute is mandatory. 
- On Metric update (or PUT) when interval is null it is deleted.

### CODE REVIEW

File: hawkular-inventory-json-helper/src/test/java/org/hawkular/inventory/json/SerializationTest.java [hunk 1]
```diff
@@ -184,7 +184,7 @@ public void testDetypedMetricType() throws Exception {
         MetricType mt = new MetricType(CanonicalPath.fromString("/t;t/mt;c"), MetricUnit.BYTES, GAUGE,
                 new HashMap<String, Object>() {{
                     put("a", "b");
-                }});
+                }}, 0L);
 
         testDetyped(mt, "{\"path\":\"/t;t/mt;c\",\"properties\":{\"a\":\"b\"},\"unit\":\"BYTES\"}");
```
Comment (metlos) [inline hawkular-inventory-json-helper/src/test/java/org/hawkular/inventory/json/SerializationTest.java]: Shouldn't the interval manifest itself in the json, too?

File: hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/Metric.java [hunk 2]
```diff
@@ -35,6 +35,7 @@
 public final class Metric extends Entity<Metric.Blueprint, Metric.Update> {
 
     private final MetricType type;
+    private final Long interval;
```
Comment (metlos) [inline hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/Metric.java]: I'd maybe be more explicit here and actually call this `collectionInterval`. "Interval of a metric" sounds a bit ambiguous to me. I'd intuitively understand it as the min-max interval of values, not a collection interval.

Comment (pavolloffay) [inline hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/Metric.java]: Good idea, collectionInterval is more explicit.

**Comment by pavolloffay** [comment]
Renamed.
I have squashed it into 1 commit since I have changed the names in both classes.

File: hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/MetricType.java [hunk 3]
```diff
@@ -41,6 +41,9 @@
     @XmlAttribute
     private final MetricDataType type;
 
+    @XmlAttribute
+    private final Long collectionInterval;
```
Comment (pilhuhn) [inline hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/MetricType.java:45]: What is the unit?

**Comment by pilhuhn** [comment]
Does / will setting this have any effect on H-metrics or the agent?

**Comment by pavolloffay** [comment]
It is in seconds, I think sync with agent is in Lukas's pipeline.

**Pull request merged**